ext/liquid_c/tokenizer.c

```diff
@@ -141,16 +141,20 @@ static void tokenizer_next_for_template(tokenizer_t *tokenizer, token_t *token)
         token->lstrip = token->rstrip;
         token->rstrip = 0;
         if (c == '%') {
+            bool has_body = false;
             while (cursor < last) {
-                if (*cursor++ != '%')
+                if (*cursor != '%') {
+                    has_body = true;
+                    cursor++;
                     continue;
+                }
                 c = *cursor++;
                 while (c == '%' && cursor <= last)
                     c = *cursor++;
                 if (c != '}')
                     continue;
                 token->type = TOKEN_TAG;
-                if(cursor[-3] == '-')
+                if (has_body && cursor[-3] == '-')
                     token->rstrip = tokenizer->lstrip_flag = true;
                 goto found;
             }
```

Comment:
We also need to fix the code path for variables.  If you add `assert(token->len_trimmed > 0);` after it has been computed at the end of the function, then that will catch the problem for variables with a `Liquid::Template.parse("{{-}}")` test.